pkgs/by-name/wl/wlink/package.nix

```
@@ -0,0 +1,50 @@
+{
+  lib,
+  rustPlatform,
+  fetchCrate,
+  pkg-config,
+  libusb1,
+  udev,
+  nix-update-script,
+  testers,
+  wlink,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "wlink";
+  version = "0.0.9";
+
+  src = fetchCrate {
+    inherit pname version;
+    hash = "sha256-Jr494jsw9nStU88j1rHc3gyQR1jcMfDIyQ2u0SwkXt0=";
+  };
+
+  cargoHash = "sha256-rPiSEfRFESYxFOat92oMUABvmz0idZu/I1S7I3g5BgY=";
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = [
+    libusb1
+    udev
+  ];
+
+  passthru = {
+    updateScript = nix-update-script { };
+    tests.version = testers.testVersion {
+      package = wlink;
+      version = "${version}";
+    };
+  };
+
+  meta = with lib; {
+    description = "WCH-Link flash tool for WCH's RISC-V MCUs(CH32V, CH56X, CH57X, CH58X, CH59X, CH32L103, CH32X035, CH641, CH643)";
+    homepage = "https://crates.io/crates/wlink";
```

No big deal about the homepage, the github repo just seems a better landing page to explore the project. The changelog would be nice though

```suggestion
    homepage = "https://github.com/ch32-rs/wlink";
    changelog = "https://github.com/ch32-rs/wlink/releases/tag/v${version}";
```

---

Yeah, that does seem like a better landing page.